wasp/apps/word_clock.py

```diff
@@ -0,0 +1,193 @@
+    ICON = icons.app
+
+    def foreground(self):
+        """Activate the application.
+
+        Configure the status bar, redraw the display and request a periodic
+        tick callback every second.
+        """
+        wasp.system.bar.clock = False
+        self._draw(True)
+        wasp.system.request_tick(1000)
+
+    def sleep(self):
+        """Prepare to enter the low power mode.
+
+        :returns: True, which tells the system manager not to automatically
+                  switch to the default application before sleeping.
+        """
+        return True
+
+    def wake(self):
+        """Return from low power mode.
+
+        Time will have changes whilst we have been asleep so we must
+        udpate the display (but there is no need for a full redraw because
+        the display RAM is preserved during a sleep.
+        """
+        self._draw()
+
+    def tick(self, ticks):
+        """Periodic callback to update the display."""
+        self._draw()
+
+    def _draw(self, redraw=False):
+        """Draw or lazily update the display.
+
+        The updates are as lazy by default and avoid spending time redrawing
+        if the time on display has not changed. However if redraw is set to
+        True then a full redraw is be performed.
+        """
+        draw = wasp.watch.drawable
+        hi =  wasp.system.theme('bright')
+        lo =  wasp.system.theme('mid')
+        mid = draw.lighten(lo, 1)
+
+        if redraw:
+            now = wasp.watch.rtc.get_localtime()
+
+            # Clear the display and draw that static parts of the watch face
+            draw.fill()
+ 
+            # Redraw the status bar
+            wasp.system.bar.draw()
+        else:
+            # The update is doubly lazy... we update the status bar and if
+            # the status bus update reports a change in the time of day 
+            # then we compare the minute on display to make sure we 
+            # only update the main clock once per minute.
+            now = wasp.system.bar.update()
+            if not now or self._min == now[4]:
+                # Skip the update
+                return
+        draw.set_color(hi)
+
+        # Format the month as text
+        month = now[1] - 1
+        month = MONTH[month*3:(month+1)*3]
+        # Record the minute that is currently being displayed
+        self._hour = now[3]
+        self._min = now[4]
+
+        # Testing
+        # self._hour = 23
+        # self._min = 59
+        
+        # Convert to words
+        part_day = ""
+        hour = ""
+        part_hour = ""
+        minute_words= ""
+
+        part_day = ""
+
+        hours_a = ["Midnight", "One", "Two", 
+                "Three", "Four", "Five", 
+                "Six", "Seven", "Eight", 
+                "Nine", "Ten", "Eleven", 
+                "Twelve",
+                "One", "Two", 
+                "Three", "Four", "Five", 
+                "Six", "Seven", "Eight", 
+                "Nine", "Ten", "Eleven"]
```

Comment:
Why it this capitalized. Is it possible to capitalize the first word of the phrase instead?